src/main/java/io/github/soc/directories/Util.java

```diff
@@ -112,63 +137,49 @@ static String linuxExecutableDir(String homeDir, String dataDir) {
-
-    // Deal with legacy or safe handling of quotes by the JDK.
-    // Safe handling may be enabled for JDKs >= 1.8.0_231, under some conditions.
-    // See https://www.oracle.com/technetwork/java/javase/8u231-relnotes-5592812.html#JDK-8221858
-    // or https://github.com/AdoptOpenJDK/openjdk-jdk8u/commit/048eb42afa11ac217dcdb690d5b266fcb910771f
-    final String q = doubleEscapeQuotes ? "\\\"" : "\"";
-
     int guidsLength = guids.length;
     StringBuilder buf = new StringBuilder(guidsLength * 68);
     for (int i = 0; i < guidsLength; i++) {
-      buf.append("[Dir]::GetKnownFolderPath(" + q);
+      buf.append("[Dir]::GetKnownFolderPath(\"");
       buf.append(guids[i]);
-      buf.append(q + ")\n");
+      buf.append("\")\n");
     }
 
+    String encodedCommand = SCRIPT_START_BASE64 + toUTF16LEBase64(buf.toString() + "}");
+
     return runCommands(guidsLength, Charset.forName("UTF-8"),
         "powershell.exe",
         "-Command",
-        "& {\n" +
-            "[Console]::OutputEncoding = [System.Text.Encoding]::UTF8\n" +
-            "Add-Type @" + q + "\n" +
-            "using System;\n" +
-            "using System.Runtime.InteropServices;\n" +
-            "public class Dir {\n" +
-            "   [DllImport(" + q + "shell32.dll" + q + ")]\n" +
-            "   private static extern int SHGetKnownFolderPath([MarshalAs(UnmanagedType.LPStruct)] Guid rfid, uint dwFlags, IntPtr hToken, out IntPtr pszPath);\n" +
-            "   public static string GetKnownFolderPath(string rfid) {\n" +
-            "       IntPtr pszPath;\n" +
-            "       if (SHGetKnownFolderPath(new Guid(rfid), 0, IntPtr.Zero, out pszPath) != 0) return " + q + q + ";\n" +
-            "       string path = Marshal.PtrToStringUni(pszPath);\n" +
-            "       Marshal.FreeCoTaskMem(pszPath);\n" +
-            "       return path;\n" +
-            "   }\n" +
-            "}\n" +
-            q + "@\n" +
-            buf.toString() +
-            "}"
+        "-EncodedCommand",
+        encodedCommand
     );
   }
 
+  private static String toUTF16LEBase64(String script) {
+    byte[] scriptInUtf16LEBytes = script.getBytes(Charset.forName("UTF-16LE"));
+    if (base64EncodeMethod == null) {
+      initBase64Encoding();
+    }
+    try {
+      return (String) base64EncodeMethod.invoke(base64Encoder, scriptInUtf16LEBytes);
+    } catch (Exception e) {
+      throw new RuntimeException("Base64 encoding failed!", e);
+    }
+  }
+
+  private static void initBase64Encoding() {
+    try {
+      base64Encoder = Class.forName("java.util.Base64").getMethod("getEncoder").invoke(null);
+      base64EncodeMethod = base64Encoder.getClass().getMethod("encodeToString", byte[].class);
+    } catch (Exception e1) {
+      try {
+        base64EncodeMethod = Class.forName("sun.misc.BASE64Encoder").getMethod("encode", byte[].class);
+      } catch (Exception e2) {
+        throw new RuntimeException("Could not find any viable Base64 encoder!", e2.initCause(e1));
```

Comment:
You don't construct `e2`, so initialising its cause isn't safe.

Indeed it seems (when I ran it on JDK 11) ok for `NoSuchMethodException` but not for `ClassNotFoundException`:

```scala
scala> val cause = new Exception("cause")
cause: Exception = java.lang.Exception: cause

scala> try Class.forName("boom") catch { case e: Exception => throw e.initCause(cause) }
java.lang.IllegalStateException: Can't overwrite cause with java.lang.Exception: cause
  at java.base/java.lang.Throwable.initCause(Throwable.java:462)
  at .liftedTree1$1(<console>:1)
  ... 28 elided
Caused by: java.lang.ClassNotFoundException: boom
  at scala.reflect.internal.util.AbstractFileClassLoader.findClass(AbstractFileClassLoader.scala:71)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
  at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
  at java.base/java.lang.Class.forName0(Native Method)
  at java.base/java.lang.Class.forName(Class.java:315)
  ... 29 more

scala> try Class.forName("java.lang.String").getMethod("boom") catch { case e: Exception => throw e.initCause(cause) }
java.lang.NoSuchMethodException: java.lang.String.boom()
  at java.base/java.lang.Class.getMethod(Class.java:2108)
  at .liftedTree1$1(<console>:1)
  ... 28 elided
Caused by: java.lang.Exception: cause
  ... 24 more
```

Reply:
Oh, well :-( ... then I guess adding some additional info to the message has to be enough (`addSuppressed` doesn't exist on older versions)

Reply:
@dwijnand Fixed.